src/main/java/com/hydratereminder/command/hydrate/HydrateCommandHandler.java

```diff
@@ -1,15 +1,32 @@
 package com.hydratereminder.command.hydrate;
 
-import javax.inject.Singleton;
-
+import com.hydratereminder.HydrateReminderPlugin;
+import com.hydratereminder.chat.ChatMessageSender;
 import com.hydratereminder.command.CommandHandler;
 
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
 @Singleton
 public class HydrateCommandHandler implements CommandHandler {
 
+    @Inject
+    ChatMessageSender chatMessageSender;
+
+    @Inject
+    HydrateReminderPlugin hydrateReminderPlugin;
+
+    /**
+     * <p>Handle the hydrate "hydrate" command by resetting the current hydrate interval, increasing
+     * hydration breaks taken during the session and displaying a hydration success message in chat
+     * </p>
+     */
     @Override
     public void handle() {
-        // TODO: Extract logic from HydrateReminderPlugin
+        hydrateReminderPlugin.hydrateBetweenHydrationBreaks();
+        hydrateReminderPlugin.setResetState(true);
```

Comment:
Are we staying with this methods in hydrateReminderPlugin as it is or should we handle this in other way?

Reply:
@macslowiak, this should be good for now as we're only refactoring the chat command logic for now.

This can be refactored further in the future after the chat command refactoring is completed.